### PR TITLE
tplay: init at 0.45

### PR DESCRIPTION
**Pull request by Demine0**
## tplay: init at 0.45
Terminal Media Player
homepage: https://github.com/maxcurzi/tplay#terminal-media-player

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/tp/tplay/package.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  src = fetchFromGitHub {
+    owner = "maxcurzi";
+    repo = pname;
+    rev =  "v${version}";
+    hash = "sha256-qt5I5rel88NWJZ6dYLCp063PfVmGTzkUUKgF3JkhLQk=";
+  };
+
+  cargoHash = "sha256-jgUKGV2Yg8+iF2wQZd1Z+QFfyJmywVTFVECUs+TK8zA=";
+  cargoPatches = [ ./cargo.diff ];
```
Comment (ambroisie) [inline pkgs/by-name/tp/tplay/package.nix:25]: Any explanations for the 2000+ lines patch?

File: pkgs/by-name/tp/tplay/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  ];
+
+  env.LIBCLANG_PATH = "${libclang.lib}/lib";
+
+  meta = {
+    description = "Terminal Media Player";
+    homepage = "https://github.com/maxcurzi/tplay";
+    platforms = lib.platforms.linux;
+    license = with lib.licenses; [ mit ];
+    maintainers = with lib.maintainers; [  ];
```
Comment (MikaelFangel) [inline pkgs/by-name/tp/tplay/package.nix:48]: Who should maintain this package? (:

Comment (MikaelFangel) [inline pkgs/by-name/tp/tplay/package.nix:48]:
```suggestion
    maintainers = with lib.maintainers; [  ];
    mainProgram = "tplay";
```

**Review by MikaelFangel — CHANGES_REQUESTED** [review]

File: pkgs/by-name/tp/tplay/package.nix (hunk 1 above)
Comment (Demine0) [reply pkgs/by-name/tp/tplay/package.nix:25]: this is a cargo.lock, i think to import this but i make it works

File: pkgs/by-name/tp/tplay/package.nix (hunk 2 above)
Comment (Demine0) [reply pkgs/by-name/tp/tplay/package.nix:48]: I could do it, but I don't understand much about nix yet, much less nixpkgs

**Review by 3JlOy-PYCCKUi — CHANGES_REQUESTED** [review]

File: pkgs/by-name/tp/tplay/package.nix (hunk 2 above)
Comment (MikaelFangel) [reply pkgs/by-name/tp/tplay/package.nix:48]: That not true. That behaviour is deprecated. See the comment on line 149: [https://github.com/NixOS/nixpkgs/blob/master/lib/meta.nix](https://github.com/NixOS/nixpkgs/blob/master/lib/meta.nix)

**Comment by Demine0** [comment]
New pr* https://github.com/NixOS/nixpkgs/pull/266300

**Pull request closed without merging**